Tests/SwiftFusionTests/Inference/GenericFactorTests.swift

```diff
@@ -19,57 +19,87 @@ import XCTest
 import PenguinStructures
 @testable import SwiftFusion
 
-/// A factor that selects a pose prior using an integer input index.
-fileprivate struct DemoFactor<Pose: LieGroup>: GenericFactor {
-  typealias Variables = Tuple2<Int, Pose>
+/// A factor that specifies a prior on a pose.
+fileprivate struct PriorFactor<Pose: LieGroup>: GenericFactor {
+  typealias Variables = Tuple1<Pose>
 
   let edges: Variables.Indices
 
-  let priors: [Pose]
+  let prior: Pose
 
   func error(_ variables: Variables) -> Double {
-    let (priorIndex, position) = (variables.head, variables.tail.head)
-    return priors[priorIndex].localCoordinate(position).squaredNorm
+    let actualPosition = variables.head
+    return prior.localCoordinate(actualPosition).squaredNorm
+  }
+}
+
+/// A factor that switches between different linear motions based on an integer label.
+fileprivate struct SwitchingLinearMotionFactor<Pose: LieGroup>: GenericFactor {
```

Comment:
Love it! But I think this is not a linear model. So, `SwitchingMotionModelFactor` ?

Reply:
Done